tests/e2e/txgen/txgen.go

```diff
@@ -0,0 +1,169 @@
+func SignAndSubmitTx(ctx context.Context, rtc client.RuntimeClient, signer signature.Signer, tx types.Transaction) error {
+	// Get chain context.
+	chainCtx, err := GetChainContext(ctx, rtc)
+	if err != nil {
+		return err
+	}
+
+	// Get current nonce for the signer's account.
+	ac := accounts.NewV1(rtc)
+	nonce, err := ac.Nonce(ctx, client.RoundLatest, types.NewAddress(signer.Public()))
+	if err != nil {
+		return err
+	}
+
+	// Sign the transaction.
+	tx.AppendSignerInfo(signer.Public(), nonce)
+	stx := tx.PrepareForSigning()
+	stx.AppendSign(chainCtx, signer)
+
+	// Submit the signed transaction.
+	if _, err = rtc.SubmitTx(ctx, stx.UnverifiedTransaction()); err != nil {
+		return err
+	}
+	return nil
+}
+
+// CreateAndFundAccount creates a new account and funds it using the
+// given funding account.
+func CreateAndFundAccount(ctx context.Context, rtc client.RuntimeClient, funder signature.Signer, id int, isSecp256k1 bool, fundAmount uint64) (signature.Signer, error) {
+	// Create new account.
+	var sig signature.Signer
+	if isSecp256k1 {
+		pk, err := btcec.NewPrivateKey(btcec.S256())
+		if err != nil {
+			return nil, err
+		}
+		sig = secp256k1.NewSigner(pk.Serialize())
+	} else {
+		cs := coreMemSig.NewTestSigner(fmt.Sprintf("test account %d", id))
+		sig = ed25519.WrapSigner(cs)
+	}
+
+	// Give it some coins.
+	tx := types.NewTransaction(&types.Fee{
+		Gas: defaultGasAmount,
+	}, "accounts.Transfer", struct {
+		To     types.Address   `json:"to"`
+		Amount types.BaseUnits `json:"amount"`
+	}{
+		To:     types.NewAddress(sig.Public()),
+		Amount: types.NewBaseUnits(*quantity.NewFromUint64(fundAmount), types.NativeDenomination),
+	})
+	if err := SignAndSubmitTx(ctx, rtc, funder, *tx); err != nil {
+		return nil, err
+	}
+
+	return sig, nil
+}
+
+// Generate generates and submits a random transaction for the given accounts
+// every txDelay seconds until the context is terminated.
+func Generate(ctx context.Context, rtc client.RuntimeClient, accounts []signature.Signer, txGens []GenerateTx, txDelay time.Duration) (error, uint64, uint64, uint64) {
+	if len(txGens) == 0 {
+		return fmt.Errorf("no transaction generators specified"), 0, 0, 0
+	}
+
+	if len(accounts) == 0 {
+		return fmt.Errorf("no accounts specified"), 0, 0, 0
+	}
+
+	if txDelay.Milliseconds() < 100 {
+		return fmt.Errorf("tx delay is too small"), 0, 0, 0
+	}
+
+	ticker := time.NewTicker(txDelay)
+	defer ticker.Stop()
+
+	var (
+		genErrCount uint64
+		subErrCount uint64
+		okCount     uint64
+	)
+
+	for {
+		select {
+		case <-ctx.Done():
+			return nil, genErrCount, subErrCount, okCount
+		case <-ticker.C:
+			// Choose random account and txn generator.
+			acct := accounts[rand.Intn(len(accounts))]
+			gen := txGens[rand.Intn(len(txGens))]
+
+			go func(acct signature.Signer, gen GenerateTx) {
+				// Generate random transaction or perform random query.
+				if tx, err := gen(ctx, rtc, acct, accounts); err != nil {
+					atomic.AddUint64(&genErrCount, 1)
```

Comment:
I had that at first, but there was too much spam -- since the transactions are generated in parallel, there are a lot of invalid nonce errors and since the generated keys are random, a lot of them don't exist in storage, etc.
It would be better to keep a histogram of error messages instead and return that.